### PR TITLE
Iron kinetic devel

### PR DESCRIPTION
**Pull request by ahundt**
Merge master with kinetic patch:
https://github.com/ThomasTimm/ur_modern_driver/issues/58
https://github.com/iron-ox/ur_modern_driver/tree/iron-kinetic-devel

May want to simply put thes into a new kinetic-devel branch until the restructured update is ready, for that reason edits from maintainers are enabled.

It at least compiles successfully, not yet tested actually running on a ur5.

### CODE REVIEW

**Comment by jacknlliu** [comment]
This PR did not work for my UR5 robot.

**Comment by ahundt** [comment]
ah that's unfortunate :-/

File: src/ur_ros_wrapper.cpp [hunk 1]
```diff
@@ -344,7 +344,7 @@ class RosWrapper {
 
 		reorder_traj_joints(goal.trajectory);
 		
-		if (!start_positions_match(goal.trajectory, 0.01)) {
+		if (!start_positions_match(goal.trajectory, 0.04)) {
```
Comment (v4hn) [inline src/ur_ros_wrapper.cpp]: Could you please change this to use a ros parameter with default `0.04` instead?
That way the user can adjust the limit in launch files if required.

For our arm, we had to increase this limit above `.08` and depending on the state of the arm other groups might need to increase it further.

**Comment by v4hn** [comment]
We don't use the `ros_control` interface.
But otherwise this request allows us to compile and use this package with our UR5 arm.

Thank you @jonbinney , @ahundt  .
Please see my inline comment for a small request.

@ThomasTimm The original [request for kinetic](https://github.com/ThomasTimm/ur_modern_driver/issues/58) is more than a year old by now. *Pretty Please* make upstream compatible with ROS kinetic. If there are further issues with the `ros_control` integration, the community will figure them out once the module is officially supposed to support kinetic.
If you require it, we can also add a bit of preprocessor magic to make this patch compatible with kinetic & indigo in the same `master` branch.

**Comment by ahundt** [comment]
https://github.com/ThomasTimm/ur_modern_driver/pull/120 was just submitted and claims to be a full refactor supporting kinetic!

**Comment by gavanderhoorn** [comment]
Seeing as #120 got merged I believe we can close this.

If you feel that is not the case please re-open and clarify.

Thanks for the PR, that is always appreciated.

**Comment by ahundt** [comment]
sounds good, thanks!

**Pull request closed without merging**